### PR TITLE
Discord destination doc

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

> Describe the changes you are proposing in this PR.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/discord.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Discord**.
+2. Assign a name to the destination and click **Continue**.
+
+### Connection settings
+
+To successfully configure Discord as a destination, you need to configure the following settings:
+
+- **Identify Template**:
+- **Event Name**: Enter the event name for which you want the event template to be triggered.
+- **Event Template**:
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:30]: Can you this complete this please.

File: docs/destinations/streaming-destinations/discord.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+A sample `identify` call is shown below:
+
+```javascript
+rudderanalytics.identify("1hKOmRA4GRlm", {
+  name: "Alex Keener",
+  email: "alex@example.com",
+  country: "USA"
+});
+```
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:48]: Message to discord:
`User Alex Keener with email: alex@example.com, country: USA is Identified`

File: docs/destinations/streaming-destinations/discord.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+### Property mappings
+
+The following table lists the mappings between the RudderStack and Discord properties:
+
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:53]: These property mappings tables are not correct can you refer to the following ones. Also there is no mapping traits being converted to a message and then that message is sent to discord.


Ruddersatck Property | Discord Property | Type |  
-- | -- | -- | --
Format from dashboard to have all the properties and name | content?* | string | Message contents (up to 2000 characters)
Message From Rudderstack | embeds?* | array of https://discord.com/developers/docs/resources/channel#embed-object objects | Embedded rich content (up to 6000 characters)

File: docs/destinations/streaming-destinations/discord.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  category: "category",
+  label: "label",
+  value: 120
+});
+```
+
+### Property mappings
+
+The following table lists the mappings between the RudderStack and Discord properties:
+
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:85]: same as with identify call

File: docs/destinations/streaming-destinations/discord.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+A sample `track` call is shown below:
+
+```javascript
+rudderanalytics.track("Order Completed", {
+  category: "category",
+  label: "label",
+  value: 120
+});
+```
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:80]: message to discord:
`Anonymous user anon_id did Order Completed with category: category label: label value: 120`

File: docs/destinations/streaming-destinations/discord.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+### Connection settings
+
+To successfully configure Discord as a destination, you need to configure the following settings:
+
+- **Identify Template**: Enter a template for building a text message while sending an `identify` call to Discord. If not provided, RudderStack uses the default template as: `User {{name}} with {{traits}} is identified`.
+- **Event Name**: Enter the event name for which you want the `track` call's event template to be triggered.
+- **Event Template**: Enter a template for building a text message while sending a `track` call to Discord. If not provided, RudderStack uses the default template as: `User {{name}} did {{event}} with {{properties}}`.
+- **Regex Matching**: Enable this setting if you want to match multiple event names in the `track` call using a regex. It helps in setting the same event template for event names matching the regex.
+- **Embed Flag**: Enable this setting to set an embed message for Discord. If enabled, also specify the following:
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:32]: `- **Embed Flag**: Enable this setting to set an embed message for Discord. If enabled, also specify the following:`
```suggestion
- **Embed Flag**: Enable this setting to set an embed message for Discord. If enabled, also specify the following:
Note :If not specified Rudderstack uses the default embed template.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+```javascript
+"content": "User Alex Keener with email: alex@example.com, country: USA is identified",
+"embeds": [{
+  "description": "Identify call made",
+  "title": "Message from Rudderstack "
+}]
+```
+
+RudderStack sends the text message after converting it into the `identify` template specified in the dashboard. Also, if the **Embed Flag** setting is enabled, it sends the text message in `content` property and the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:68]:
```suggestion
RudderStack sends `the text message` in `content` property after converting it into the `identify` template specified in the dashboard. Also, if the **Embed Flag** setting is enabled, the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+});
+```
+
+<div class="infoBlock">
+Either one of the <code class="inline-code">userId</code> or <code class="inline-code">anonymousId</code> is required to make an <code class="inline-code">identify</code> call successfully.
+</div>
+
+RudderStack sends the above `identify` call to Discord in the following text message format (using the defaut `identify` template):
+- `User Alex Keener with email: alex@example.com, country: USA is identified`
+- When **Embed Flag** setting is enabled and the **Title Template** and **Description Template** are provided in the dashboard settings:
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/discord.mdx:58]:
```suggestion
- When **Embed Flag** setting is enabled and the **Title Template** and **Description Template** are not provided i.e. default in the dashboard settings:
```

**Review by anantjain45823 — APPROVED** [review]

File: docs/destinations/streaming-destinations/discord.mdx [hunk 9]
```diff
@@ -0,0 +1,99 @@
+---
+title: "Discord"
+description: Send your event data from RudderStack to Discord.
+---
+
+[Discord](https://discord.com/) is a popular communications app used by gamers to communicate online. It lets you share voice, video, and text chat with friends, game communities, and developers.
+
+Find the open-source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/discord">GitHub repository</a>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:8]:
```suggestion
Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/discord">GitHub repository</a>.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+Once you have confirmed that the source platform supports sending events to Discord, follow these steps:
+
+1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Discord**.
+2. Assign a name to the destination and click **Continue**.
+
+### Connection settings
+
+To successfully configure Discord as a destination, you need to configure the following settings:
+
+- **Identify Template**: Enter a template for building a text message while sending an `identify` call to Discord. If not provided, RudderStack uses the default template as: `User {{name}} with {{traits}} is identified`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:28]:
```suggestion
- **Identify Template**: Enter a template for building a text message while sending an `identify` call to Discord. If not provided, RudderStack uses `User {{name}} with {{traits}} is identified` as the default template.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+- **Event Name**: Enter the event name for which you want the `track` call's event template to be triggered.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:29]:
```suggestion
- **Event Name**: Enter the event name for which RudderStack triggers the `track` call's event template.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+- **Event Template**: Enter a template for building a text message while sending a `track` call to Discord. If not provided, RudderStack uses the default template as: `User {{name}} did {{event}} with {{properties}}`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:30]:
```suggestion
- **Event Template**: Enter a template for building a text message while sending a `track` call to Discord. If not provided, RudderStack uses `User {{name}} did {{event}} with {{properties}}` as the default template.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+- **Regex Matching**: Enable this setting if you want to match multiple event names in the `track` call using a regex. It helps in setting the same event template for event names matching the regex.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:31]:
```suggestion
- **Regex Matching**: Enable this setting if you want to match multiple event names in the `track` call using a regex. It helps in setting the same event template for the event names matching the regex.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+- **Embed Flag**: Enable this setting to set an embed message for Discord. If enabled, also specify the following:
+    - **Title Template**: Enter the template for building the title of the embed message. If not provided, RudderStack uses the default title as: `Identify/Track call made`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:33]:
```suggestion
    - **Title Template**: Enter the template for building the title of the embed message. If not provided, RudderStack uses `Identify/Track call made` as the default title.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+    - **Description Template**: Enter the template for building the description of the embed message. If not provided, RudderStack uses the default description as: `Message from RudderStack`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:34]:
```suggestion
    - **Description Template**: Enter the template for building the description of the embed message. If not provided, RudderStack uses `Message from RudderStack` as the default description .
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+- **Webhook Url**: Enter the webhook URL of a Discord channel where you want to send the data. You can create it by logging in to [Discord](https://discord.com/login) and selecting the channel for which you want to send the data. Further, navigate to the **Channel settings** > **Integrations** > **Create Webhook**.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:35]:
```suggestion
- **Webhook Url**: Enter the webhook URL of the Discord channel where you want to send the data. You can create it by logging in to [Discord](https://discord.com/login), selecting the channel, and going to the **Channel settings** > **Integrations** > **Create Webhook**.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+  email: "alex@example.com",
+  country: "USA"
+});
+```
+
+<div class="infoBlock">
+Either one of the <code class="inline-code">userId</code> or <code class="inline-code">anonymousId</code> is required to make an <code class="inline-code">identify</code> call successfully.
+</div>
+
+RudderStack sends the above `identify` call to Discord in the following text message format (using the defaut `identify` template):
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:56]:
```suggestion
RudderStack sends the above `identify` call to Discord in the following text message format (using the default `identify` template):
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+- When the **Embed Flag** setting is enabled and the default values for **Title Template** and **Description Template** are used:
+
+```javascript
+"content": "User Alex Keener with email: alex@example.com, country: USA is identified",
+"embeds": [{
+  "description": "Identify call made",
+  "title": "Message from Rudderstack "
+}]
+```
+RudderStack sends the text message in `content` property after converting it into the `identify` template specified in the dashboard. Also, if the **Embed Flag** setting is enabled, it sends the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:67]:
```suggestion
RudderStack sends the text message in the `content` property after converting it into the `identify` template specified in the dashboard. Also, if the **Embed Flag** setting is enabled, it sends the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+```javascript
+"content": "Anonymous user anon_id did Order Completed with category: category, label: label, value: 120",
+"embeds": [{
+  "description": "Track call made",
+  "title": "Message from Rudderstack "
+}]
+```
+
+RudderStack sends the text message in `content` property after converting it into the `track` template specified in the dashboard. Also, if the **Embed Flag** setting is enabled, it sends the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:99]:
```suggestion
RudderStack sends the text message in `content` property after converting it into the `track` template specified in the **Event Template** dashboard setting. Also, if the **Embed Flag** setting is enabled, it sends the embed message details in the `embeds` property. Refer to the [Discord documentation](https://discord.com/developers/docs/resources/webhook) for more information.
```

File: docs/destinations/streaming-destinations/discord.mdx [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+  label: "label",
+  value: 120
+});
+```
+
+<div class="infoBlock">
+<code class="inline-code">event name</code> is a required property to make a <code class="inline-code">track</code> call successfully.
+</div>
+
+RudderStack sends the above `track` call to Discord in the following text message format (using the defaut `track` template):
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:87]:
```suggestion
RudderStack sends the above `track` call to Discord in the following text message format (using the default `track` template specified in the **Event Template** dashboard setting):
```

Comment (ishachopra0401) [reply docs/destinations/streaming-destinations/discord.mdx:87]: default template is taken when no template is defined in the dashboard setting.

File: docs/destinations/streaming-destinations/discord.mdx [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/discord.mdx:87]:
```suggestion
RudderStack sends the above `track` call to Discord in the following text message format (using the default `track` template):
```

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**